skglm/solvers/base.py

```diff
@@ -1,8 +1,26 @@
-from abc import abstractmethod
+from abc import abstractmethod, ABC
+from skglm.utils.validation import check_obj_solver_attr
 
 
-class BaseSolver():
-    """Base class for solvers."""
+class BaseSolver(ABC):
+    """Base class for solvers.
+
+    Attributes
+    ----------
+    _datafit_required_attr : list
+        List of attributes that must be implemented in Datafit.
+
+    _penalty_required_attr : list
+        List of attributes that must be implemented in Penalty.
+
+    Notes
+    -----
+    For required attributes, if an attribute is given as a list of attributes
```

Comment:
Can you be a bit more explciit like : "if an element of the required attribute list is itself a list, it means that at least of theses attributes must be implemented
    e.g. `_datafit_required_attr = ("a", ("b", "c"))` means that the datafit must implement `a`, together with at least one of `b` and `c`"  ?

Reply:
Yes indeed